### DISCUSSION TITLE
Issue retrieving hierarchy nodes with `power_bi.get_member_properties` on an alternative hierarchy

### DISCUSSION THREAD

**Discussion by Sven-Bo** [cubewise-code/tm1py#866] [Q&A] [answered]
The "TM1py Region" dimension has two hierarchies: the default hierarchy "TM1py Region" and the alternative hierarchy "TM1py City". Here is a screenshot of my "TM1py Region" dimension:
![dimension](https://filedn.eu/lV0RcQd1Re5jQr5XL93Ntwm/tm1py_issue/tm1py_dimension.png)

When I use the default hierarchy, "TM1py Region", the levels are present in the resulting dataframe. This is the code I used:

```python
df = tm1.power_bi.get_member_properties(
    dimension_name="TM1py Region",
    hierarchy_name="TM1py Region",
)
```
![dataframe_default_hierarchy](https://filedn.eu/lV0RcQd1Re5jQr5XL93Ntwm/tm1py_issue/tm1py_dimension_hierarchy.png)

However, when I use the power_bi.get_member_properties function to retrieve hierarchy nodes from the "TM1py City" hierarchy, the levels are not returned in the resulting dataframe. This is the code I used:

```python
df = tm1.power_bi.get_member_properties(
    dimension_name="TM1py Region",
    hierarchy_name="TM1py City",
)
```
And this is the resulting dataframe:
![dataframe_alt_hierarchy](https://filedn.eu/lV0RcQd1Re5jQr5XL93Ntwm/tm1py_issue/tm1py_dimension_alt_hierarchy.png)

I'm wondering if this is expected behaviour and how I can retrieve the levels when using an alternative hierarchy.

Thank you for your help.

**Comment by MariusWirtz** [comment]
Hi @Sven-Bo,

thanks for raising this issue.
I was able to replicate it using the code below.

I will try to provide a fix in the next days.

``` python
from TM1py.Objects.Dimension import Dimension

from TM1py import TM1Service, ElementAttribute
from TM1py.Objects.Element import Element
from TM1py.Objects.Hierarchy import Hierarchy

tm1_local = {
    "address": "localhost",
    "port": 8010,
    "user": "admin",
    "password": "apple",
    "ssl": True
}

tm1 = TM1Service(**tm1_local)

hierarchy1 = Hierarchy(
    name="TM1py Region",
    dimension_name="TM1py Region",
    elements=[Element("Europe", "Consolidated"), Element("CH", "Numeric"), Element("UK", "Numeric")],
    element_attributes=[ElementAttribute("Currency", "String")],
    edges={("Europe", "CH"): 1, ("Europe", "UK"): 1})

hierarchy2 = Hierarchy(
    name="TM1py City",
    dimension_name="TM1py Region",
    elements=[Element("Germany", "Consolidated"), Element("Munich", "Numeric"), Element("Munich", "Numeric")],
    edges={("Germany", "Munich"): 1, ("Germany", "Munich"): 1})

dimension = Dimension(name="TM1py Region", hierarchies=[hierarchy1, hierarchy2])

tm1.dimensions.update_or_create(dimension)

df = tm1.power_bi.get_member_properties(dimension_name="TM1py Region", hierarchy_name="TM1py City")
print(df)
```

**Reply by FranciscoLeiteAlves** [reply]
I have an identical problem, I can't use tm1.power_bi.get_member_properties to retrieve dimension data
df = tm1.power_bi.get_member_properties(dimension_name="TM1py Region", hierarchy_name="TM1py City")
TM1py.Exceptions.Exceptions.TM1pyRestException: Text: '{"error":{"code":"278","message":"'ExecuteMDXSetExpression' resource can not be resolved on type 'EntityContainer'."}}' - Status Code: 404 - Reason: 'Not Found' - Headers: {'Content-Length': '124', 'Connection': 'keep-alive', 'Content-Encoding': 'gzip', 'Cache-Control': 'no-cache', 'Content-Type': 'application/json; charset=utf-8', 'OData-Version': '4.0'}

**Reply by FranciscoLeiteAlves** [reply]
Would the version of Tm1 be installed?

**Reply by gbryant-dev** [reply, marked as answer]
@FranciscoLeiteAlves Not sure how the problem is identical here as you're getting an error. However, from the error message, it looks like you're not using a sufficient TM1 version as `get_member_properties` ultimately calls the `ExecuteMDXSetExpression` action in the TM1 REST API which requires TM1 version 11.1.0 or PA 2.0.3.
